### PR TITLE
BaSP-TG-78: Update admin forms

### PR DESCRIPTION
**Pull request by LuliBianchini**
### Tasks done in the PR:
- Implementation of React Hook Form in Edit and Add Admins Forms
- Implementation of Joi validations in both forms 

### Expected result:
- Correct visualization and working of forms with their corresponding inputs

### CODE REVIEW

File: src/Components/Admins/Add/index.js [hunk 1]
```diff
@@ -1,100 +1,133 @@
...
+  firstName: Joi.string()
+    .min(3)
+    .max(50)
+    .pattern(/^[\p{L}\p{M}]*$/u)
+    .required()
+    .messages({
+      'string.min': 'Invalid name, it must contain more than 3 letters',
+      'string.max': 'Invalid name, it must not contain more than 50 letters',
+      'string.pattern.base': 'Invalid name, it must contain only letters',
+      'any.required': 'First Name is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Add/index.js:20]: In order to display this response "First Name is a required field", you need to change the 'any.required' to 'string.empty'

File: src/Components/Admins/Add/index.js [hunk 2]
```diff
@@ -1,100 +1,133 @@
...
+  lastName: Joi.string()
+    .min(3)
+    .max(50)
+    .pattern(/^[\p{L}\p{M}]*$/u)
+    .required()
+    .messages({
+      'string.min': 'Invalid last name, it must contain more than 3 letters',
+      'string.max': 'Invalid last name, it must not contain more than 50 letters',
+      'string.pattern.base': 'Invalid last name, it must contain only letters',
+      'any.required': 'Last Name is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Add/index.js:31]: The same as above

File: src/Components/Admins/Add/index.js [hunk 3]
```diff
@@ -1,100 +1,133 @@
...
+      'any.required': 'Email is a required field'
+    }),
+  password: Joi.string()
+    .min(8)
+    .pattern(/^(?=.*?[a-zA-Z])(?=.*?[0-9])(?!.*[^a-zA-Z0-9])/)
+    .required()
+    .messages({
+      'string.min': 'Invalid password, it must contain at least 8 characters',
+      'string.pattern.base': 'Invalid password, it must contain both letters and numbers',
+      'any.required': 'Password is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Add/index.js:47]: The same as above

File: src/Components/Admins/Add/index.js [hunk 4]
```diff
@@ -1,100 +1,133 @@
...
+      'string.max': 'Invalid last name, it must not contain more than 50 letters',
+      'string.pattern.base': 'Invalid last name, it must contain only letters',
+      'any.required': 'Last Name is a required field'
+    }),
+  email: Joi.string()
+    .email({ tlds: { allow: false } })
+    .required()
+    .messages({
+      'string.email': 'Invalid email format',
+      'any.required': 'Email is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Add/index.js:38]: The same as above

File: src/Components/Admins/Edit/index.js [hunk 5]
```diff
@@ -1,114 +1,144 @@
...
+  firstName: Joi.string()
+    .min(3)
+    .max(50)
+    .pattern(/^[\p{L}\p{M}]*$/u)
+    .required()
+    .messages({
+      'string.min': 'Invalid name, it must contain more than 3 letters',
+      'string.max': 'Invalid name, it must not contain more than 50 letters',
+      'string.pattern.base': 'Invalid name, it must contain only letters',
+      'any.required': 'First Name is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Edit/index.js:20]: The same as above

File: src/Components/Admins/Edit/index.js [hunk 6]
```diff
@@ -1,114 +1,144 @@
...
+      'any.required': 'Email is a required field'
+    }),
+  password: Joi.string()
+    .min(8)
+    .pattern(/^(?=.*?[a-zA-Z])(?=.*?[0-9])(?!.*[^a-zA-Z0-9])/)
+    .required()
+    .messages({
+      'string.min': 'Invalid password, it must contain at least 8 characters',
+      'string.pattern.base': 'Invalid password, it must contain both letters and numbers',
+      'any.required': 'Password is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Edit/index.js:47]: The same as above

File: src/Components/Admins/Edit/index.js [hunk 7]
```diff
@@ -1,114 +1,144 @@
...
+      'string.max': 'Invalid last name, it must not contain more than 50 letters',
+      'string.pattern.base': 'Invalid last name, it must contain only letters',
+      'any.required': 'Last Name is a required field'
+    }),
+  email: Joi.string()
+    .email({ tlds: { allow: false } })
+    .required()
+    .messages({
+      'string.email': 'Invalid email format',
+      'any.required': 'Email is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Edit/index.js:38]: The same as above

File: src/Components/Admins/Edit/index.js [hunk 8]
```diff
@@ -1,114 +1,144 @@
...
+  lastName: Joi.string()
+    .min(3)
+    .max(50)
+    .pattern(/^[\p{L}\p{M}]*$/u)
+    .required()
+    .messages({
+      'string.min': 'Invalid last name, it must contain more than 3 letters',
+      'string.max': 'Invalid last name, it must not contain more than 50 letters',
+      'string.pattern.base': 'Invalid last name, it must contain only letters',
+      'any.required': 'Last Name is a required field'
```
Comment (NicolasFontana) [inline src/Components/Admins/Edit/index.js:31]: The same as above

**Review by NicolasFontana — CHANGES_REQUESTED** [review]
There are only small details, the rest looks good!

**Review by NicolasFontana — APPROVED** [review]
Looks nice! Good work

**Comment by LeonelListro** [comment]
# Test results

Tested locally on: https://alexis-trackgenix-app-git-feature-tg-78-updat-4d17f8-basp-m2022.vercel.app/admins

test result: Approved ✅

**Review by LeonelListro — APPROVED** [review]

**Pull request merged**